### DISCUSSION TITLE
SNX upgrade problem

### DISCUSSION THREAD

**Discussion by itiloo** [ruyrybeyro/chrootvpn#30] [Q&A]
Hi Ruy,
Thank you very much for your script.
I upgrade a R80.40 with last jumbo hotfix in our checkpoint gateways and now the script vpn.sh dont function.

vpn.sh status command say

SNX - installed              800008304
SNX - available for download 800008407
CShell - installed version      80,0,0070,28
CShell - available for download 80,0,0070,49

Can I upgrade snx and chsell version without reinstall?
If I do "vpn.sh uninstall" and do "vpn.sh -i --vpn=url", that reinstall a download the new snx version and all is ok, but I wanted to know if there was any way to update it.

Thank you so much.

**Comment by ruyrybeyro** [comment]
Hi, 

It can be done, not much time in the next weeks for changing the script.

Manually, snx_install.sh and cshell_install.sh have to be loaded to /opt/chroot/root ,

and then run:

vpn.sh stop
vpn.sh shell

Afterwards, inside the chroot (with cshell stopped, per the vpn.sh stop):

/root/snx_install.sh
PATH=/nopatch:"${PATH}" /root/cshell_install.sh
exit 0

Regards

**Reply by ruyrybeyro** [reply]
@itiloo Made some edits to the bash code of the previous answer. Per scripts in the /nopatch area, xhost does nothing when running chell_install.sh, so DISPLAY is not really needed. Also code gets more compact moving PATH to the invocation line of the script.

**Comment by itiloo** [comment]
Thanks Rui:

Then, the steps are,

1.-Copy new snx_install.sh and cshell_install.sh to /opt/chroot/root ,

2.-vpn.sh stop & vpn.sh shell

3.-/root/snx_install.sh

4.-export PATH=/nopatch:"${PATH}"

5.-/root/cshell_install.sh

Is this correct?.

It would be nice if when it started (vpn.sh start), it would automatically
check if there is a new version, and if there is, it would download both
files and do it automatically.
I'm going to try to implement it in the startup script.
I think in the end it takes less time (when supporting a company user),
telling him to reinstall by doing "vpn.sh uninstall" and "vpn.sh -i
--vpn=dnsname", this is faster believe.

Sorry for my bad english, I am spanish.

**Reply by ruyrybeyro** [reply]
Yeah, that's right, followed by exit for leaving the chroot shell.

One of the issues with automating this process is that since the script runs at login, there's uncertainty about when it's appropriate to download the necessary files. Introducing tests and upgrades can lead to complications and delays, and in some cases, might even result in damaged chroot environments.

**Comment by itiloo** [comment]
In the script I would add that it does a "vpn.sh status" looking for the
installed snx version and compares it with the available snx version, and
only if they are different do the process.

**Reply by ruyrybeyro** [reply]
Fair enough, it might work.

**Comment by itiloo** [comment]
It´s works perfect Rui,
Thank you so much.

Your script is wonderful and it saves me a lot of problems with VPN users.
The natural installation form that checkpoint has is rubbish, I think they
have it very abandoned and they don't care much about Linux.

Thanks you so much.
